crates/red_knot_python_semantic/resources/mdtest/suppressions/no-type-check.md

```diff
@@ -0,0 +1,94 @@
+# `@no_type_check`
+
+> If a type checker supports the `no_type_check` decorator for functions, it should suppress all type errors for the def statement and its body including any nested functions or classes. It should also ignore all parameter and return type annotations and treat the function as if it were unannotated.
+> [source](https://typing.readthedocs.io/en/latest/spec/directives.html#no-type-check)
+
+## Error in the function body
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int: 
+    return a + 5
+```
+
+## Error in nested function
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int:
+    def nested(): 
+        return a + 5
+```
+
+## Error in nested class
+
+```py
+from typing import no_type_check
+
+@no_type_check
+def test() -> int:
+    class Nested:
+        def inner(self):
+            return a + 5
+```
+
+## Error in decorator
+
+Both MyPy and Pyright flag the `unknown_decorator` but we don't. 
```

Comment:
Thanks for the extra input. I intentionally left this open for now because the approach on how to implement this probably depends on how we handle those diagnostics :) I somewhat agree but then again, `no_type_check` disables type checking for the entire function... which I would expect includes all decorators.

Reply:
WIth later: I mainly mean, once we figured out how to implement this properly 😆